source/common/stats/symbol_table_impl.cc

```diff
@@ -215,6 +215,13 @@ Symbol SymbolTable::toSymbol(absl::string_view sv) {
     }
   }
 
+  // TODO(jmarantz): we could use __builtin_expect here to avoid any production
```

Comment:
I'm not sure that branch prediction guidance would yield no overhead unless I misunderstand how it works. Thinking about it more because, why not, I bet you could do this with no overhead via some type of template definition of the class. Essentially, allow the test hooks to be specified via template. This should allow a null specialization to be empty and compiled out vs. the test specialization. This isn't a big deal but it's a fun puzzle. :)